File: api/lib/contract/contract.ex
defmodule AppWeb.Models.Contract do
  @derive [Poison.Encoder, Jason.Encoder]
  defstruct [
    :id,
    :user,
    :address,
    :balance,
    :currency,
    :type,
    :config,
    :date,
    :amountInUsd
  ]
end

File: ui/src/model/Contract.ts

export interface AbiItem {
  constant: boolean;
  inputs: Array<any>;
  name: string;
  outputs: Array<any>;
  payable: boolean;
  stateMutability: string;
  type: string;
}

export interface LotteryConfig {
  ticketPrice: number;
}

export interface GuessNumberConfig {
  numberToGuess: number;
  ticketPrice: number;
}

export interface DonationConfig {
  minValue: number;
}

export type ContractType = 'donation' | 'guess' | 'lottery';

export default interface Contract {
  id: string;
  address?: string;
  user: {
    name: string;
    surname?: string;
    email?: string;
  },
  date?: number;
  currency?: string;
  deployPriceInUsd?: number;
  amountInUsd?: number;
  balance?: number;
  type?: ContractType;
  config: LotteryConfig | GuessNumberConfig | DonationConfig | any;
};
